pkg/actuator/actuator.go

```diff
@@ -0,0 +1,201 @@
+}
+
+func (a *Actuator) GenerateSecretData(logger logr.Logger, ctx context.Context, ex *extensionsv1alpha1.Extension,
+	chartPath string, namespace string, k8sversion string, configs map[string]string) (map[string][]byte, error) {
+	emptyMap := map[string][]byte{}
+	// Depending on shoot, chartredner will have different capabilities based on K8s version.
+	chartRenderer, err := a.ChartRendererFactory.NewChartRendererForShoot(k8sversion)
+	if err != nil {
+		return emptyMap, err
+	}
+	imageVector := imagevector.ImageVector()
+	if len(imageVector) > 0 {
+		for i, imageSource := range imageVector {
+
+			logger.Info(fmt.Sprintf("imageVector[%d].imageSource", i), "imageSource", *imageSource)
+		}
+	}
+
+	installationImage, err := imageVector.FindImage(consts.InstallationImageName)
+	if err != nil {
+		return emptyMap, err
+	}
+	agentImage, err := imageVector.FindImage(consts.AgentImageName)
+	if err != nil {
+		return emptyMap, err
+	}
+	chartValues := map[string]interface{}{
+		"images": map[string]string{
+			consts.InstallationImageName: installationImage.String(),
+			consts.AgentImageName:        agentImage.String(),
+		},
+		"configs": configs,
+	}
+	release, err := chartRenderer.Render(chartPath, consts.InstallationReleaseName, metav1.NamespaceSystem, chartValues)
+	//release, err := chartRenderer.RenderEmbeddedFS(chartPath, InstallationReleaseName, metav1.NamespaceSystem, chartValues)
+
+	if err != nil {
+		return emptyMap, err
+	}
+	// Put chart into secret
+	secretData := map[string][]byte{consts.InstallationSecretKey: release.Manifest()}
+	return secretData, nil
+}
+
+func (a *Actuator) Reconcile(ctx context.Context, logger logr.Logger, ex *extensionsv1alpha1.Extension) error {
+	namespace := ex.GetNamespace()
+	a.logger.Info("Reconcile: checking extension...") // , "shoot", cluster.Shoot.Name, "namespace", cluster.Shoot.Namespace)
+
+	// Find what shoot cluster we dealing with.
+	// to find k8s version for chart renderer
+	// and get providerConfig for configurations for CRI-resource-manager configmaps
+	// with imageVector support this would allow to choose different version of extensions depending on k8s version
+	cluster, err := extensionscontroller.GetCluster(ctx, a.client, namespace)
+	if err != nil {
+		return err
+	}
+
+	// Get configs either from providerConfig and merged it with provided files
+	configs, err := configs.GetConfigs(a.logger, cluster.Shoot.Spec.Extensions)
+	if err != nil {
+		panic(err)
+		// return err
+	}
+
+	secretData, err := a.GenerateSecretData(logger, ctx, ex, consts.ChartPath, namespace, cluster.Shoot.Spec.Kubernetes.Version, configs)
+	if err != nil {
+		panic(err)
+		// return err
+	}
+
+	// Reconcile managedresource and secret for shoot.
+	if err := managedresources.CreateForShoot(ctx, a.client, namespace, consts.ManagedResourceName, false, secretData); err != nil {
+		return err
+	}
+
+	// Don't reconcile if already there ?! or we want to update with new configs/scripts?
```

Comment:
To be answered before merging. If there is a new config/script it would make sense to reconcile.

Reply:
Yes I agree -  I moved all this considerations to issue when we can improve/optimise reconilation loop - the code is not best place to keeps such code snippets

Here you have: https://github.com/intel/gardener-extension-cri-resmgr/issues/43

but anyway done by removing